playwright/_impl/_frame.py

```diff
@@ -276,9 +276,25 @@ async def wait_for_selector(
         strict: bool = None,
         timeout: float = None,
         state: Literal["attached", "detached", "hidden", "visible"] = None,
-    ) -> Optional[ElementHandle]:
-        return from_nullable_channel(
-            await self._channel.send("waitForSelector", locals_to_params(locals()))
+    ) -> ElementHandle:
+        return cast(
+            ElementHandle,
+            from_nullable_channel(
```

Comment:
```suggestion
            from_channel(
```

No need for `cast` here